pkg/observer/controller.go

```diff
@@ -309,6 +370,15 @@ func (c *controller) Run() {
 	klog.V(3).Infof("waiting for %v to allow changes to settle", c.WaitInterval)
 	select {
 	case <-time.After(c.WaitInterval):
+		timeout := 0
+		retries := 20
+		for !c.safeToStartCycle() {
+			time.Sleep(10 * time.Second)
```

Comment:
Perhaps query using exponential backoff github.com/cenkalti/backoff.

Reply:
Rather we'd use `client-go` provided backoff package instead
`https://github.com/kubernetes/client-go/blob/master/util/flowcontrol/backoff.go`